lib/events/s3sessions/s3handler.go

```diff
@@ -372,31 +388,35 @@ func (h *Handler) uploadFile(ctx context.Context, path string, reader io.Reader)
 // result into a writer. Returns trace.NotFound error if the recording is not
 // found.
 func (h *Handler) Download(ctx context.Context, sessionID session.ID, writer events.RandomAccessWriter) error {
-	return h.downloadFile(ctx, h.recordingPath(sessionID), writer)
+	return trace.Wrap(h.downloadOriginalFile(ctx, h.recordingPath(sessionID), writer))
 }
 
 // DownloadSummary downloads a session summary from an S3 bucket and writes the
 // results into a writer. Returns trace.NotFound error if the summary is not
 // found.
 func (h *Handler) DownloadSummary(ctx context.Context, sessionID session.ID, writer events.RandomAccessWriter) error {
-	return h.downloadFile(ctx, h.summaryPath(sessionID), writer)
+	err := h.downloadFile(ctx, h.summaryPath(sessionID), writer, nil /* versionID */)
+	if trace.IsNotFound(err) {
+		return trace.Wrap(h.downloadFile(ctx, h.pendingSummaryPath(sessionID), writer, nil /* versionID */))
```

Comment:
Is the pending summary removed when a summary completes?

If so, do we have a potential race here where the first download fails because the summary hasn't finished, and then the summary finishes (and the pending summary is removed) before we attempt to download the pending summary?